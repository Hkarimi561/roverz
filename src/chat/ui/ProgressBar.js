import React from 'react';
import {
  ProgressViewIOS,
  Platform,
  Text,
} from 'react-native';

export default class ProgressBar extends React.Component {
  constructor(props) {
    super(props);
    const progress = this.props.progress;
    this.state = {
      progress,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      progress: nextProps.progress,
    });
  }

  render() {
    if (Platform.OS === 'ios') {
      return (
        <ProgressViewIOS
          progress={this.state.progress}
          progressTintColor={'#43D35D'}
          trackTintColor={'#DCDCDC'}
        />
      );
    }
    return (<Text>Loading</Text>);
  }
}

ProgressBar.defaultProps = {
  progress: 0.5,
};

ProgressBar.propTypes = {
  progress: React.PropTypes.number,
};
