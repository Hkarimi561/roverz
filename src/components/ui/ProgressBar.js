import React from 'react';
import {
  ProgressViewIOS,
  Platform,
  Text,
} from 'react-native';
import PropTypes from 'prop-types';

import t from '../../i18n';


export default class ProgressBar extends React.Component {
  constructor(props) {
    super(props);
    const progress = this.props.progress;
    this.state = {
      progress,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      progress: nextProps.progress,
    });
  }

  render() {
    if (Platform.OS === 'ios') {
      return (
        <ProgressViewIOS
          progress={this.state.progress}
          progressTintColor={'#43D35D'}
          trackTintColor={'#DCDCDC'}
        />
      );
    }
    return (<Text>{t('txt_loading')}</Text>);
  }
}

ProgressBar.defaultProps = {
  progress: 0.5,
};

ProgressBar.propTypes = {
  progress: PropTypes.number,
};
