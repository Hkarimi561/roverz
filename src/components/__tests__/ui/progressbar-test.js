/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Platform } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProgressBar from '@ui/ProgressBar';

it('ProgressBar renders correctly', () => {
  const tree = renderer.create(
    <ProgressBar />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('progressbar componentWillReceiveProps', () => {
  const progressComponent = renderer.create(<ProgressBar />).getInstance();
  const tree = progressComponent.componentWillReceiveProps(50);
  expect(progressComponent.state.progress).toEqual(tree);
});

it('android', () => {
  Platform.OS = 'android';
  const component = renderer.create(<ProgressBar />).toJSON();
  expect(component).toMatchSnapshot();
});
it('ios', () => {
  Platform.OS = 'ios';
  const component = renderer.create(<ProgressBar />).toJSON();
  expect(component).toMatchSnapshot();
});
