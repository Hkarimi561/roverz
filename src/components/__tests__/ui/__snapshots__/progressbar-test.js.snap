// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `
<RCTProgressView
  progress={0.5}
  progressTintColor="#43D35D"
  style={
    Array [
      Object {
        "height": 2,
      },
      undefined,
    ]
  }
  trackTintColor="#DCDCDC"
/>
`;

exports[`android 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Loading
</Text>
`;

exports[`ios 1`] = `
<RCTProgressView
  progress={0.5}
  progressTintColor="#43D35D"
  style={
    Array [
      Object {
        "height": 2,
      },
      undefined,
    ]
  }
  trackTintColor="#DCDCDC"
/>
`;
